### ISSUE TITLE
app hash  not match bug

### ISSUE THREAD

**Issue by youngqqcn** [orientwalt/htdf#15]
## Environment:

- 4 nodes ( A,  B , C , D ) (internal)
- solidity file  `htdf_faucet.sol` 
- `upgrade ` protocol version v2 

## Steps:

- deploy contract
- deposit some HTDF 
- call  `getOneHtdf` 
- call `setOnceAmount` by non-owner
- call `getOneHtdf` by non-owner

send many normal transactions(about 2000 txs) at the same time .

## Log

when `getOneHtdf` be included in block `N`,  the chain stop at `N + 1`, and log `proposal block is invalid ... appHash ...`

node A and C :  expected appHash and  got appHash are samely 
node B and D :  expected appHash and  got appHash are samely 

but A C  is different with B D 

Node A:
```
E[2020-12-08|19:33:48.920] enterPrevote: ProposalBlock is invalid       module=consensus height=902 round=38 err="Wrong Block.Header.AppHash.  Expected 0A4E0E943B3503032E425610B884E89037B795C8CB1F11D54036A569B40EC19C, got E6F82FD08245FC03DBA8E4BCA886E9E9C7C85E67002DE86F8B38779473191AF4"
E[2020-12-08|19:34:49.619] Connection failed @ sendRoutine              module=p2p peer=7292fe1c7dcf921f2176a9a178610dba7d233ac8@192.168.0.174:44082 conn=MConn{192.168.0.174:44082} err="pong timeout"
E[2020-12-08|19:34:49.619] Stopping peer for error                      module=p2p peer="Peer{MConn{192.168.0.174:44082} 7292fe1c7dcf921f2176a9a178610dba7d233ac8 in}" err="pong timeout"
E[2020-12-08|19:35:53.333] enterPrevote: ProposalBlock is invalid       module=consensus height=902 round=41 err="Wrong Block.Header.AppHash.  Expected 0A4E0E943B3503032E425610B884E89037B795C8CB1F11D54036A569B40EC19C, got E6F82FD08245FC03DBA8E4BCA886E9E9C7C85E67002DE86F8B38779473191AF4"

```

Node B:
```
[2020-12-08|19:31:17.130] enterPrevote: ProposalBlock is invalid       module=consensus height=902 round=34 err="Wrong Block.Header.AppHash.  Expected E6F82FD08245FC03DBA8E4BCA886E9E9C7C85E67002DE86F8B38779473191AF4, got 0A4E0E943B3503032E425610B884E89037B795C8CB1F11D54036A569B40EC19C"
E[2020-12-08|19:32:31.100] enterPrevote: ProposalBlock is invalid       module=consensus height=902 round=36 err="Wrong Block.Header.AppHash.  Expected E6F82FD08245FC03DBA8E4BCA886E9E9C7C85E67002DE86F8B38779473191AF4, got 0A4E0E943B3503032E425610B884E89037B795C8CB1F11D54036A569B40EC19C"

```

Node C:

```
consensus height=902 round=11 err="Wrong Block.Header.AppHash.  Expected 0A4E0E943B3503032E425610B884E89037B795C8CB1F11D54036A569B40EC19C, got E6F82FD08245FC03DBA8E4BCA886E9E9C7C85E67002DE86F8B38779473191AF4"
E[2020-12-08|19:22:37.757] enterPrevote: ProposalBlock is invalid       module=consensus height=902 round=14 err="Wrong Block.Header.AppHash.  Expected 0A4E0E943B3503032E425610B884E89037B795C8CB1F11D54036A569B40EC19C, got E6F82FD08245FC03DBA8E4BCA886E9E9C7C85E67002DE86F8B38779473191AF4"

```

Node D had been crashed  when I reappeared the bug again. Mempool was full, it contains 5000 txs ,  and it did not process transactions any more.

**Comment by youngqqcn** [comment]
I have reappeared this bug again just now .  protocol version is 0  (v1.2.0)
```
$hsd version
GitCommit=95cdeb6|version=1.2.0|GitBranch=fix_apphash_bug|
```

![appHashBug1](https://user-images.githubusercontent.com/30211020/101569145-ddcbb480-3a0e-11eb-91d3-35ec26db8a3e.png)

**Comment by youngqqcn** [comment]
fixed at v1.3.1

https://github.com/orientwalt/htdf/commit/1ab1ea5c7e3d3857363661f0b07be25e18178375
